lib/srv/desktop/rdp/rdpclient/src/rdpdr/mod.rs

```diff
@@ -702,6 +728,52 @@ impl Client {
         }
     }
 
+    fn process_irp_read(
+        &mut self,
+        device_io_request: DeviceIoRequest,
+        payload: &mut Payload,
+    ) -> RdpResult<Vec<Vec<u8>>> {
+        // https://github.com/FreeRDP/FreeRDP/blob/511444a65e7aa2f537c5e531fa68157a50c1bd4d/channels/drive/client/drive_main.c#L268
+        let rdp_req = DeviceReadRequest::decode(device_io_request, payload)?;
+        debug!("received RDP: {:?}", rdp_req);
+
+        if let Some(file) = self.file_cache.remove(rdp_req.device_io_request.file_id) {
+            self.tdp_sd_read(rdp_req, file)
+        } else {
+            self.prep_read_response(rdp_req, NTSTATUS::STATUS_UNSUCCESSFUL, vec![])
+        }
+    }
+
+    fn process_irp_write(
+        &mut self,
+        device_io_request: DeviceIoRequest,
+        payload: &mut Payload,
+    ) -> RdpResult<Vec<Vec<u8>>> {
+        let rdp_req = DeviceWriteRequest::decode(device_io_request, payload)?;
+        debug!("received RDP: {:?}", rdp_req);
+
+        if let Some(file) = self.file_cache.remove(rdp_req.device_io_request.file_id) {
+            self.tdp_sd_write(rdp_req, file)
+        } else {
+            self.prep_write_response(rdp_req, NTSTATUS::STATUS_UNSUCCESSFUL, 0)
+        }
+    }
+
+    fn process_irp_set_information(
+        &mut self,
+        device_io_request: DeviceIoRequest,
+        payload: &mut Payload,
+    ) -> RdpResult<Vec<Vec<u8>>> {
+        let rdp_req = ServerDriveSetInformationRequest::decode(device_io_request, payload)?;
+
+        // TODO(LKozlowski): handle it?
```

Comment:
handle what?

Reply:
We need to handle the request and then generate appropriate response when we'll be implementing the `TDP Move` message. Right now we just decode it and then generate reply to make server not crash. I wouldn't be able to test write requests and responses without it as the server sends this request to change last modified attribute on write. I've updated the TODO message so we know what it is about and we don't forget about it.